src/SpectatorView.Unity/Assets/SpectatorView/Scripts/MarkerDetection/QRCode/QRCodesManager.cs

```diff
@@ -1,302 +1,387 @@
+                            location = Matrix4x4.TRS(translation, rotation, Vector3.one);
+
+                            return true;
+                        }
+                        else
+                        {
+                            Debug.LogWarning("QRCode location unknown or not yet available.");
+                            return false;
+                        }
+                    }
+                    else
+                    {
+                        Debug.LogWarning("Failed to obtain coordinate system for application");
+                        return false;
+                    }
+                }
+                catch(Exception e)
+                {
+                    Debug.LogWarning($"Note: TryGetLocationForQRCode needs to be called from main thread: {e}");
+                    return false;
+                }
+            }
+            else
+            {
+                Debug.LogWarning("Failed to obtain coordinate system for QRCode");
+                return false;
+            }
+        }
+#endif // WINDOWS_UWP
+
+        public Guid GetIdForQRCode(string qrCodeData)
+        {
+            lock (qrCodesList)
+            {
+                foreach (var ite in qrCodesList)
+                {
+                    if (ite.Value.Code == qrCodeData)
+                    {
+                        return ite.Key;
+                    }
+                }
+            }
+            return new Guid();
+        }
+
+        public IList<QRCodesTrackerPlugin.QRCode> GetList()
+        {
+            lock (qrCodesList)
+            {
+                return new List<QRCodesTrackerPlugin.QRCode>(qrCodesList.Values);
+            }
+        }
+
+        private void Awake()
+        {
+            IsTrackerRunning = false;
+        }
+
+        protected async void Start()
+        {
+            if (AutoStartQRTracking)
+            {
+                await StartQRTrackingAsync();
+            }
+        }
+
+        public Task<QRTrackerStartResult> StartQRTrackingAsync(CancellationToken cancellationToken = default)
+        {
+            lock (lockObj)
+            {
+                if (startTrackerTask != null)
+                {
+                    DebugLog("Returning existing start tracker task");
+                    return startTrackerTask;
+                }
+
+                stopTrackerTask = null;
+                startTrackerCTS = new CancellationTokenSource();
+
+                var hybridCTS = CancellationTokenSource.CreateLinkedTokenSource(startTrackerCTS.Token, cancellationToken);
+                DebugLog("Kicking off a new start tracker task");
+                return startTrackerTask = Task.Run(() => StartQRTrackingAsyncImpl(hybridCTS.Token), hybridCTS.Token);
+            }
+        }
+
+        private async Task<QRTrackerStartResult> StartQRTrackingAsyncImpl(CancellationToken token)

```

Comment:
>token [](start = 92, length = 5)

Should this cancellation token be used anywhere? A lot of work is going in to creating the right kind of linked token to never use it :-)

Reply:
we check the cancellation token before starting now